dfdaemon/proxy/proxy.go

```diff
@@ -179,6 +179,7 @@ func (proxy *Proxy) mirrorRegistry(w http.ResponseWriter, r *http.Request) {
 	t, err := transport.New(
 		transport.WithDownloader(proxy.downloadFactory()),
 		transport.WithTLS(proxy.registry.TLSConfig()),
+		transport.WithCondition(proxy.shouldUseDfgetForRegistryMirror),
```

Comment:
Since you've added the registry mirror to proxy rules in #929 , you only have to use a very simple condition here.
```go
transport.WithCondition(func(req *http.Request) bool {
    // TODO: handle nil
    return !proxy.registry.Direct
})
```

Reply:
I consider that multi-mirrors may be supported in the future, so I match the host.
We do not need that indeed currently.

Reply:
> I consider that multi-mirrors may be supported in the future, so I match the host.

How about we only support a single registry for now as a start, and add support in for multiple registries in another PR when we've decided how to handle multiple registries in #945 ?

@zcc35357949 @lowzj @Starnop

Reply:
> > I consider that multi-mirrors may be supported in the future, so I match the host.
> 
> How about we only support a single registry for now as a start, and add support for multiple registries in another PR when we've decided how to handle multiple registries in #945 ?
> 
> @zcc35357949 @lowzj @Starnop

ok, I have simplified the condition func.